src/lib.rs

```diff
@@ -557,25 +608,22 @@ impl<'a> AggregatedQueries<'a> {
                     .map(|rule_listener_query| rule_listener_query.resolve(language.language()))
                     .enumerate()
                 {
+                    let capture_index_if_per_capture: CaptureIndexIfPerCapture =
+                        match &rule_listener_query.match_by {
+                            ResolvedMatchBy::PerCapture { capture_index } => Some(*capture_index),
+                            _ => None,
+                        };
+
                     for _ in 0..rule_listener_query.query.pattern_count() {
-                        per_language_builder
-                            .pattern_index_lookup
-                            .push((rule_index, rule_listener_index));
+                        per_language_builder.pattern_index_lookup.push((
+                            rule_index,
+                            rule_listener_index,
+                            capture_index_if_per_capture,
+                        ));
                     }
-                    let query_text_with_unified_capture_name =
```

Comment:
I didn't realize this at first but now we can completely get rid of this "unifying" of the individual-listener specified capture names in the big "aggregated" query (I think)